spinn_front_end_common/interface/interface_functions/data_in_multicast_routing_generator.py

```diff
@@ -0,0 +1,219 @@
+    FAKE_ETHERNET_CHIP_Y = 0
+    ROUTING_MASK = 0xFFFFFFFC
+
+    def __call__(self, machine, extra_monitor_cores, placements,
+                 board_version):
+        # create progress bar
+        progress = ProgressBar(
+            machine.ethernet_connected_chips,
+            "Generating routing tables for data in system processes")
+
+        # create routing table holder
+        routing_tables = MulticastRoutingTables()
+        key_to_destination_map = dict()
+
+        for ethernet_chip in progress.over(
+                machine.ethernet_connected_chips):
+            fake_graph, fake_placements, fake_machine, \
+                key_to_destination_map_new = self._create_fake_network(
+                    ethernet_chip, machine, extra_monitor_cores,
+                    placements, board_version)
+
+            # update dict for key mapping
+            key_to_destination_map.update(key_to_destination_map_new)
+
+            # do routing
+            routing_tables_by_partition = self.do_routing(
+                fake_graph=fake_graph, fake_placements=fake_placements,
+                fake_machine=fake_machine)
+            self._generate_routing_tables(
+                routing_tables, routing_tables_by_partition, ethernet_chip,
+                machine)
+        return routing_tables, key_to_destination_map
+
+    def _generate_routing_tables(
+            self, routing_tables, routing_tables_by_partition, ethernet_chip,
+            machine):
+        """ from the routing. use the partition id as key, and build mc\
+         routing tables
+
+        :param routing_tables: the routing tables to store routing tables in
+        :param routing_tables_by_partition: the routing output
+        :param ethernet_chip: the ethernet chip being used
+        :param machine: the SpiNNMachine instance
+        :return: dict of chip x and chip yto key to get there
+        :rtype: dict
+        """
+        for fake_chip_x, fake_chip_y in \
+                routing_tables_by_partition.get_routers():
+            partitions_in_table = routing_tables_by_partition.\
+                get_entries_for_router(fake_chip_x, fake_chip_y)
+
+            real_chip_x, real_chip_y = \
+                helpful_functions.calculate_machine_level_chip_id(
+                    fake_chip_x, fake_chip_y, ethernet_chip.x, ethernet_chip.y,
+                    machine)
+
+            multicast_routing_table = MulticastRoutingTable(
+                real_chip_x, real_chip_y)
+
+            for partition in partitions_in_table:
+
+                # build routing table entries
+                entry = partitions_in_table[partition]
+                multicast_routing_table.add_multicast_routing_entry(
+                    MulticastRoutingEntry(
+                        routing_entry_key=partition.identifier,
+                        mask=DataInMulticastRoutingGenerator.ROUTING_MASK,
+                        processor_ids=entry.out_going_processors,
```

Comment:
How did this ever work?

Reply:
That property is called `processor_ids`…

…But you changed it in commit SpiNNakerManchester/PACMAN@4632a459af872c5ccef17b574abbd6b98060cf54…

…Which post-dates this PR. Oh.